components/json-splitter.js

```diff
@@ -0,0 +1,88 @@
+// json-splitter.js
+
+/**
+ * This component takes a json string with one or more array elements, parses it, and makes each array element into an 
+ * individual records.  Each record is adorned with a virtual node ID (vnid) to uniquely identify it in the rdf pipeline, 
+ * and sends them on to the next component.
+ */
+
+var jsonpointer = require('jsonpointer');
+
+var _ = require('underscore');
+var baseNode = require('./base-node');
+
+exports.getComponent = baseNode.updatePromise({
+
+  description: 'Splits a json dataset into individual records, adds a vnid to each, and sends them on.',
+
+  inPorts: { 
+      in: { 
+        addressable: true, 
+        datatype: 'string',
+        description: 'json string input',
+        required: true
+      },
+
+      json_pointer: {
```

Comment:
It will receive the data on the 'in' port, the json_pointer is the attribute name (or path) that contains the vnid, such as 'key' or 'id'